src/transformers/models/bartpho/tokenization_bartpho.py

```diff
@@ -301,10 +311,29 @@ def save_vocabulary(self, save_directory: str, filename_prefix: Optional[str] =
             (filename_prefix + "-" if filename_prefix else "") + VOCAB_FILES_NAMES["monolingual_vocab_file"],
         )
 
-        if os.path.abspath(self.vocab_file) != os.path.abspath(out_vocab_file):
+        if os.path.abspath(self.vocab_file) != os.path.abspath(out_vocab_file) and os.path.isfile(self.vocab_file):
             copyfile(self.vocab_file, out_vocab_file)
-
-        if os.path.abspath(self.monolingual_vocab_file) != os.path.abspath(out_monolingual_vocab_file):
+        elif not os.path.isfile(self.vocab_file):
+            with open(out_vocab_file, "wb") as fi:
+                content_spiece_model = self.sp_model.serialized_model_proto()
+                fi.write(content_spiece_model)
+
+        if os.path.abspath(self.monolingual_vocab_file) != os.path.abspath(
+            out_monolingual_vocab_file
+        ) and os.path.isfile(self.monolingual_vocab_file):
             copyfile(self.monolingual_vocab_file, out_monolingual_vocab_file)
+        elif not os.path.isfile(self.monolingual_vocab_file):
+            with open(out_monolingual_vocab_file, "w", encoding="utf-8") as fp:
+                for token in self.fairseq_tokens_to_ids:
+                    if token not in [
+                        str(self.bos_token),
+                        str(self.eos_token),
+                        str(self.sep_token),
+                        str(self.cls_token),
+                        str(self.unk_token),
+                        str(self.pad_token),
+                        str(self.mask_token),
+                    ]:
```

Comment:
Styling nit: remove all the `str` from here and add it in the line below, then the list should fit in one line (maybe we'll need to define it above at one or two less indentation levels).

Reply:
Thanks a lot for your review! :slightly_smiling_face: 

This is a good idea. However, I can't make it fit on one line (even if I move the variable up by 4 indentations). I propose the format in [this commit](https://github.com/huggingface/transformers/pull/15328/commits/0e89b275834bc31b21f9acc87bdec59e82084d0e). Is it ok for you?

Reply:
Works for me!

Reply:
I was a little too enthusiastic, but in fact it seems to me that one cannot do without putting the casting of the string on each `Xxx_token` because we must check here that the token is not part of the list of the special tokens - which can be some `AddedTokens` and not pure strings.

I'm reverting it in a new commit.